libqtile/core/manager.py

```diff
@@ -131,23 +135,35 @@ def __init__(
         for i in self.groups:
             self.groups_map[i.name] = i
 
-        for grp in self.config.groups:
+        for grp in config.groups:
             if isinstance(grp, ScratchPadConfig):
                 sp = ScratchPad(grp.name, grp.dropdowns, grp.label)
-                sp._configure([self.config.floating_layout],
-                              self.config.floating_layout, self)
+                sp._configure([config.floating_layout],
+                              config.floating_layout, self)
                 self.groups.append(sp)
                 self.groups_map[sp.name] = sp
 
-        self._eventloop = eventloop
-        self.setup_eventloop()
-        self.server = IPCCommandServer(self)
+        # It fixes problems with focus when clicking windows of some specific clients like xterm
+        def noop(qtile):
+            pass
+        config.mouse += (Click([], "Button1", lazy.function(noop), focus="after"),)
 
-        self.current_screen = None
-        self.screens = []
+        return config
+
+    def dump_state(self, buf):
+        try:
+            pickle.dump(QtileState(self), buf, protocol=0)
+        except:  # noqa: E722
```

Comment:
Why ignoring E722 exactly?

Reply:
I didn't add this, just moved it. Bare except is almost never acceptable... unless you're restoring a pickle =) E722 is that linting error.

Reply:
Oh right, didn't see it was there already. Why do you say it's acceptable for pickles?

Reply:
They can throw literally anything
https://docs.python.org/2/library/pickle.html#pickle.UnpicklingError

Reply:
Yeah but they still only raise BaseException children, don't they? They wont send us a KeyboardInterrupt or alike, and catching these is probably one of the things that currently prevents us from quitting the tests, for example. :)